quickstart/daml/licensing-tests/daml/Licensing/Scripts/TestLicense.daml

```diff
@@ -1,261 +1,545 @@
+  : (AmuletRegistry.AmuletRegistry, Party, Party)
+  -> ContractId LicenseRenewalRequest
+  -> ContractId License
+  -> Decimal
+  -> InstrumentId
+  -> Script ()
+acceptAndCompleteLicenseRenewal (registry, userParty, providerParty)
+                                renewalRequestCid
+                                licenseCid
+                                renewalFee
+                                instrumentId = script do
+
+  now <- getTime
+
+  ----------------------------------------------------------------------------
+  -- 1. The user sees the allocation request in their wallet
+  ----------------------------------------------------------------------------
+  [userAlloc] <- WalletClient.listRequestedAllocations userParty instrumentId
+  userAlloc.transferLeg.amount === renewalFee
+
+  -- The user obtains their current holding IDs
+  userHoldingCids <- WalletClient.listHoldingCids userParty instrumentId
+
+  ----------------------------------------------------------------------------
+  -- 2. The user accepts (allocates) the request by calling the registry factory
+  ----------------------------------------------------------------------------
+  enrichedChoice <- RegistryApi.getAllocationFactory registry AllocationFactory_Allocate with
+    expectedAdmin = registry.dso
+    allocation = userAlloc
+    requestedAt = now
+    inputHoldingCids = userHoldingCids
+    extraArgs = emptyExtraArgs
+
+  submitWithDisclosures' userParty enrichedChoice.disclosures $
+    exerciseCmd enrichedChoice.factoryCid enrichedChoice.arg
+
+  -- Advance time to allow the user acceptance to be finalized
+  passTime (hours 1)
+
+  ----------------------------------------------------------------------------
+  -- 3. The provider checks the allocation for the given request & finalizes
+  ----------------------------------------------------------------------------
+  -- The provider sees the single allocation in the system, with
+  -- settlementRef referencing the renewal request
+  Some renewalRequest <- queryContractId providerParty renewalRequestCid
+  let renewalRef = (view $ toInterface @AllocationRequest renewalRequest).settlement.settlementRef
+  allocations <- listAllocationsForRenewal providerParty renewalRef
+  length allocations === 1
+
+  let [(allocCid, _)] = allocations
+
+  ----------------------------------------------------------------------------
+  -- 4. The provider completes the renewal payment by finalizing the allocation
+  ----------------------------------------------------------------------------
+  context <- RegistryApi.getAllocation_TransferContext registry allocCid emptyMetadata
+  let extraArgs = ExtraArgs with
+        context = context.choiceContext
+        meta = emptyMetadata
+
+  submitWithDisclosures' providerParty context.disclosures $
+    exerciseCmd renewalRequestCid LicenseRenewalRequest_CompleteRenewal with
+      allocationCid = allocCid
+      licenseCid = licenseCid
+      extraArgs = extraArgs
+
```

Comment:
Check that the expected effects happened: License contract updated, no open Renewal, funds moved.